.github/actions/github-deployments-delete/action.yml

```diff
@@ -0,0 +1,56 @@
+name: Delete Github Deployments
+
+description:
+  Deletes all GitHub deployments on a given branch.
+  Used as workaround to delete the flood of messages visible on some PRs where environments are leveraged but deployments are not.
+
+inputs:
+  branch-name:
+    description: The name of the branch for the deployments
+    required: true
+  environment:
+    description: The optional environment for the deployments
+    required: false
+
+runs:
+  using: composite
+  steps:
+    - name: delete-deployments
+      uses: actions/github-script@60a0d83039c74a4aee543508d2ffcb1c3799cdea # v7.0.1
+      with:
+        script: |
```

Comment:
wouldn't be better to move everything to a .js and then load it?

https://github.com/actions/github-script?tab=readme-ov-file#run-a-separate-file

should make also testing locally and lint/indent file properly